### PR TITLE
Enable eyedropper

### LINKED ISSUE #8247: Enable feature: Eyedropper
## Feature Description
Eyedropper is currently behind a feature flag, original PR: https://github.com/google/web-stories-wp/pull/7961

There a few things to do:
- [x] Fix and enable the Karma test for Eyedropper
- [x] Fix [the bug that Claudio found ](https://github.com/google/web-stories-wp/issues/262#issuecomment-875986055) Edit: Not able to reproduce this and it wasn't reported in the bug bash either -- might be an edge case to follow up later. Please file a bug for it if not able to reproduce.
- [x] Create patch file for `html-to-image`, see https://github.com/google/web-stories-wp/pull/8118#discussion_r665751490
- [x] Add a spinner for when the eyedropper is loading (image being generated). https://github.com/google/web-stories-wp/issues/8261 -
- [x] [DONE IN A SEPARATE PR] https://github.com/google/web-stories-wp/issues/8262 - tiny style fix. 
- [x] Mark the feature flag `true` by default (will enable the feature but allows to remove it easily if needed for some reason).

Creating a patch (instructions from Pascal):
1. Install regular version
2. Modify files directly in `node_modules/html-to-image/..`
3. Run `npx patch-package html-to-image`

### PR DESCRIPTION
**Pull request by merapi**
## Summary

- [x] Fix and enable the Karma test for Eyedropper
- [x] Fix for Safari `html-to-image` XMLSerializer issue
- [x] Create patch file for `html-to-image`, see https://github.com/google/web-stories-wp/pull/8118#discussion_r665751490
- [x] Add a spinner for when the eyedropper is loading (image being generated). https://github.com/google/web-stories-wp/issues/8261 -
- [x] Mark the feature flag `true` by default (will enable the feature but allows to remove it easily if needed for some reason).

## User-facing changes

Enable eyedropper.

## Testing Instructions & QA

Open color picker, use the eyedropper to pick the color.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8247

### CODE REVIEW

File: assets/src/edit-story/components/canvas/eyedropperLayer.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ import styled from 'styled-components';
 import { rgba, readableColor } from 'polished';
 import { FULLBLEED_RATIO } from '@web-stories-wp/units';
 
+/**
+ * WordPress dependencies
+ */
+import { Spinner } from '@wordpress/components';
```
Comment (swissspidy) [inline assets/src/edit-story/components/canvas/eyedropperLayer.js]: We don't import any UI components from `@wordpress/`, ever

Instead import `CircularProgress` from `../circularProgress` or `LoadingSpinner` from `@web-stories-wp/design-system`

**Comment by miina** [comment]
@merapi Some tests are failing.

One additional finding testing on a 13'' screen:
eyedropper

Looks like the generated image is larger than the actual page on this smaller screen.

File: assets/src/edit-story/components/canvas/eyedropperLayer.js (hunk 1 above)
Comment (merapi) [reply assets/src/edit-story/components/canvas/eyedropperLayer.js]: Oh, right, I saw those imports but they are in `web-stories-block` 🤦🏻

**Comment by merapi** [comment]
> @merapi Some tests are failing.
> 
> One additional finding testing on a 13'' screen:
> Looks like the generated image is larger than the actual page on this smaller screen.

Test fixed.
I can't reproduce that small screen issue, is it easily reproducible for you @miina ?

https://user-images.githubusercontent.com/21036224/125819321-70879253-a3e0-4a8c-af18-d97f1c49b1d5.mov

**Comment by miina** [comment]
> I can't reproduce that small screen issue, is it easily reproducible for you @miina ?

Yep, happens every time when trying. On Chrome, full screen size (on a 13'').

**Comment by barklund** [comment]
> > I can't reproduce that small screen issue, is it easily reproducible for you @miina ?
> 
> Yep, happens every time when trying. On Chrome, full screen size (on a 13'').

I'm able to reproduce this only inconsistently when resizing the window - sometimes the image replacement flashes in the wrong size, sometimes it shows up in the right size from the start.

**Review by barklund — APPROVED** [review]
Overall this looks great. But make sure you get buyin from infra pod and that the remaining issues are solved 🎉

File: patches/html-to-image+1.6.2.patch [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+\ No newline at end of file
+diff --git a/node_modules/html-to-image/lib/index.js b/node_modules/html-to-image/lib/index.js
+index b18a4be..3ffa079 100644
+--- a/node_modules/html-to-image/lib/index.js
++++ b/node_modules/html-to-image/lib/index.js
+@@ -1,3 +1,4 @@
++alert('lib');
```
Comment (swissspidy) [inline patches/html-to-image+1.6.2.patch]: This should be removed

File: patches/html-to-image+1.6.2.patch [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+\ No newline at end of file
+diff --git a/node_modules/html-to-image/es/index.js b/node_modules/html-to-image/es/index.js
+index ddb8dda..21900a0 100644
+--- a/node_modules/html-to-image/es/index.js
++++ b/node_modules/html-to-image/es/index.js
+@@ -13,6 +13,7 @@ import { embedWebFonts, getWebFontCss } from './embedWebFonts';
+ import { createSvgDataURL } from './createSvgDataURL';
+ import { applyStyleWithOptions } from './applyStyleWithOptions';
+ import { delay, createImage, canvasToBlob, getNodeWidth, getNodeHeight, getPixelRatio, } from './util';
++console.log('ts');
```
Comment (swissspidy) [inline patches/html-to-image+1.6.2.patch]: This should be removed

File: patches/html-to-image+1.6.2.patch (hunk 2 above)
Comment (merapi) [reply patches/html-to-image+1.6.2.patch]: Nice catch, I thought nobody is looking at those messy patch files 😁

**Comment by merapi** [comment]
Small screen issues and those from Pascal are fixed, moving to QA.

**Pull request merged**